hack/docker-push.sh

```diff
@@ -71,6 +71,9 @@ if [[ ! -z "$TAG" ]]; then
 else
     echo "We're on branch $BRANCH"
     VERSION="$BRANCH"
```

Comment:
any reason not to always add `-dev` ?

Reply:
Some testing code references the `main` tag currently, doesn't append `dev` for the main branch to avoid changing the code